### PR TITLE
Antifungal drugs work over time

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary

SUMMARY: Balance "Antifungal drugs work over time"

#### Purpose of change

Antifungal drugs had instantaneous effect. Moreover they did not prevent reinfecting instantly afterwards.
Marloss local guides could gulp antifungal drugs all day and bathe in fungicide with no repercussions.

#### Describe the solution

- [x] Fungal infection was standardized as duration dependent with 6h per intensity level = 18h max duration & max intensity = 3
- [x] Antifungal drugs now have their own effects that counters fungal infection over time. One dose lasts 4 hours.
- [x] Antifungal effect reverses the progression clock of the fungal infection, and it regresses 5 times faster then it progresses. With the standardization of fungal infection the recovery process will take ~4 hours, so one dose will be sufficient. However, during that time all effects of fungal infection still take place, but the presence of anti-fungal agent in the system will significantly reduce the frequency / chance of symptoms appearance.
- [x] As long as antifungals are in effect, you actually gain relative immunity for the duration of the effect. That is unless you use fungal blooms as bongs, as inhaling spores en masse or being injected by spores multiplies them faster then the drug can kill them.
- [x] Fungicide will add shorter (1h) antifungal effect, as it is only applied externally.
- [x] Antifungal effect will now result in deadly poisoning for Marloss local guides. You wanted to be a fungus? Don't chomp on antifungal, don't bathe in fungicide. Only hope: blood filter CBM to filter out the drug before it melts you.

#### Describe alternatives you've considered

N/A

#### Testing

Reworked unit test, passes + some in-game testing.

#### Additional context

I love fungaloids as they are a single enemy that introduces authentic doom looming over the world. Every other faction seems to be winnable against, but with this enemy you battle both the visible and the unseen enemy, and the wide front of the terraforming invasion makes you a speck of dust in face of the ever moving cogs of the fungal domination. That said player has to be able to stand his ground, even just in the scope of a single person. This change by default buys the player a bit more time to find antifungal meds when infected. And while migrating instant healing to over time recovery effect seems worse, it in fact introduces a form of passing immunity, that may push the balance in favor of the player when dealing with fungaloids.

### CODE REVIEW

File: src/effect.cpp [hunk 1]
```diff
@@ -771,7 +771,7 @@ void effect::set_duration( const time_duration &dur, bool alert )
         set_intensity( duration / eff_type->int_dur_factor + 1, alert );
     }
 
-    add_msg( m_debug, "ID: %s, Duration %d", get_id().c_str(), to_turns<int>( duration ) );
+    add_msg( m_debug, "ID: %s, Duration %s", get_id().c_str(), to_string( duration ) );
```
Comment (ZhilkinSerg) [inline src/effect.cpp:774]: Further profiling showed that was not a good change - `to_string` calls translation routines which are pretty slow.

image

**Comment by nexusmrsep** [comment]
You can revert this portion. Small readability improvement over performance regression is not worth it. But on the other hand shouldn't it pop only in debug mode and not influence the game while not in debug mode?

**Comment by anothersimulacrum** [comment]
The string formatting has to happen later, and so the debug mode skips it, but the call to `to_string` cannot be deferred.

**Comment by ZhilkinSerg** [comment]
I am going to fix this without reverting in https://github.com/CleverRaven/Cataclysm-DDA/pull/43004.

**Pull request merged**